policy.go

```diff
@@ -784,95 +960,118 @@ func executePolicySession(tpm *tpm2.TPMContext, policySession tpm2.SessionContex
-	if err := tpm.PolicyNV(pinIndex, pinIndex, policySession, operandB, 0, tpm2.OpUnsignedLE, revocationCheckSession); err != nil {
-		switch {
-		case tpm2.IsTPMError(err, tpm2.ErrorPolicy, tpm2.CommandPolicyNV):
-			// The dynamic authorization policy has been revoked.
-			return dynamicPolicyDataError{errors.New("the dynamic authorization policy has been revoked")}
-		case tpm2.IsTPMSessionError(err, tpm2.ErrorPolicyFail, tpm2.CommandPolicyNV, 1):
-			// Either staticInput.PinIndexAuthPolicies is invalid or the NV index isn't what's expected, so the key file is invalid.
-			return staticPolicyDataError{errors.New("invalid PIN NV index or associated authorization policy metadata")}
+		operandB := make([]byte, 8)
+		binary.BigEndian.PutUint64(operandB, dynamicInput.policyCount)
+		if err := tpm.PolicyNV(policyCounter, policyCounter, policySession, operandB, 0, tpm2.OpUnsignedLE, revocationCheckSession); err != nil {
+			switch {
+			case tpm2.IsTPMError(err, tpm2.ErrorPolicy, tpm2.CommandPolicyNV):
+				// The PCR policy has been revoked.
+				return dynamicPolicyDataError{errors.New("the PCR policy has been revoked")}
+			case tpm2.IsTPMSessionError(err, tpm2.ErrorPolicyFail, tpm2.CommandPolicyNV, 1):
+				// Either staticInput.v0PinIndexAuthPolicies is invalid or the NV index isn't what's expected, so the key file is invalid.
+				return staticPolicyDataError{errors.New("invalid PCR policy counter or associated authorization policy metadata")}
+			}
+			return xerrors.Errorf("PCR policy revocation check failed: %w", err)
 		}
-		return xerrors.Errorf("dynamic authorization policy revocation check failed: %w", err)
 	}
 
-	authPublicKey := staticInput.AuthPublicKey
+	authPublicKey := staticInput.authPublicKey
 	if !authPublicKey.NameAlg.Supported() {
-		return staticPolicyDataError{errors.New("public area of dynamic authorization policy signature verification key has an unsupported name algorithm")}
+		return staticPolicyDataError{errors.New("public area of dynamic authorization policy signing key has an unsupported name algorithm")}
 	}
 	authorizeKey, err := tpm.LoadExternal(nil, authPublicKey, tpm2.HandleOwner)
 	if err != nil {
 		if tpm2.IsTPMParameterError(err, tpm2.AnyErrorCode, tpm2.CommandLoadExternal, 2) {
 			// staticInput.AuthPublicKey is invalid
-			return staticPolicyDataError{errors.New("public area of dynamic authorization policy signature verification key is invalid")}
+			return staticPolicyDataError{errors.New("public area of dynamic authorization policy signing key is invalid")}
 		}
-		return xerrors.Errorf("cannot load public area for dynamic authorization policy signature verification key: %w", err)
+		return xerrors.Errorf("cannot load public area for dynamic authorization policy signing key: %w", err)
 	}
 	defer tpm.FlushContext(authorizeKey)
 
+	var pcrPolicyRef tpm2.Nonce
+	if version > 0 {
+		pcrPolicyRef = computePcrPolicyRefFromCounterContext(policyCounter)
```

Comment:
this is probably a good place to repeat the reasons/purpose of policyRef

Reply:
I've added this.